### ISSUE TITLE
npm Script explorer duplicates scripts when folder is included in the workspace multiple times

### ISSUE THREAD

**Issue by forivall** [Microsoft/vscode#49237]
Issue Type: <b>Bug</b>

1. Create a workspace in a folder with a package.json with scripts
2. Add the parent folder to the workspace
3. Open up or refresh the npm script explorer

The scripts are shown twice.

<img width="194" alt="screen shot 2018-05-04 at 11 34 00 am" src="https://user-images.githubusercontent.com/760204/39646052-1106b0d8-4f8f-11e8-8429-c26c219c3eb4.png">

bug.code-workspace
```
{
	"folders": [
		{
			"path": "1"
		},
		{
			"path": "."
		}
	],
	"settings": {}
}
```
package.json
```
{
  "name": "1",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "Emily Marigold Klassen <forivall@gmail.com>",
  "license": "ISC",
  "dependencies": {}
}
```

VS Code version: Code 1.23.0 (7c7da59c2333a1306c41e6e7b68d7f0caa7b3d45, 2018-05-03T15:23:14.634Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|16.00GB (0.02GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron -psn_0_5010631|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (71)</summary>

Extension|Author (truncated)|Version
---|---|---
indent4to2|Com|0.1.2
racket|Dmy|1.2.0
xml|Dot|1.9.2
graphviz-preview|EFa|1.0.0
EditorConfig|Edi|0.12.1
gc-excelviewer|Gra|2.0.21
beautify|Hoo|1.3.0
awesometypescriptproblemmatcher|Jua|0.3.0
material-icon-theme|PKi|3.3.0
dot|Ste|0.0.1
quokka-vscode|Wal|1.0.121
sort-imports|ama|5.1.0
previewseqdiag-vscode|ari|0.1.0
vscode-refresh-html|baj|0.0.2
spellright|ban|2.3.3
vscode-icontheme-nomo-dark|be5|1.3.5
unique-lines|bib|1.0.0
markdown-checkbox|bie|0.1.1
markdown-preview-github-styles|bie|0.1.2
vscode-raml|blz|3.0.1
box-comment|can|0.2.0
npm-intellisense|chr|1.3.0
path-intellisense|chr|1.4.2
gitignore|cod|0.5.0
vscode-svgviewer|css|1.4.3
vscode-eslint|dba|1.4.8
chmod|dle|1.0.0
githistory|don|0.4.0
gitlens|eam|8.2.4
tslint|eg2|1.0.28
vscode-npm-script|eg2|0.3.4
git-project-manager|fel|1.5.1
abnf|for|1.0.0
pegjs-syntax|fut|0.1.1
rest-client|hum|0.18.4
vscode-ebnf|igo|1.0.3
launcher|ili|0.0.4
docthis|joe|0.6.0
json-escaper|jos|1.0.1
fuzzy-definitions|jri|0.0.3
racket|kar|0.0.3
rainbow-csv|mec|0.2.1
vscode-duplicate|mrm|1.2.1
mssql|ms-|1.3.1
atom-keybindings|ms-|3.0.4
cpptools|ms-|0.16.1
csharp|ms-|1.14.0
team|ms-|1.133.0
vscode-mscgen|msc|1.1.0
debugger-for-chrome|msj|4.3.0
editorconfiggenerator|nep|0.2.1
vetur|oct|0.11.7
vscode-code-outline|pat|0.1.1
open-html-in-browser|pea|0.0.3
vscode-template-literal-editor|pli|0.9.0
typescript-mono-repo-import-helper|q|0.0.5
java|red|0.24.0
vscode-sort-json|ric|1.11.0
npm-link-status|rob|0.0.3
nunjucks|ron|0.2.3
linter-xo|sam|2.1.3
vscode-open|san|0.1.0
vscode-zipexplorer|sle|0.3.1
vscode-odata|sta|0.1.0
rewrap|stk|1.8.1
html-preview-vscode|tht|0.1.1
pdf|tom|0.2.0
vscode-sourcemaps-navigator|vlk|0.0.3
vscode-apielements|vnc|0.6.7
change-case|wma|1.0.0
vscode-open-in-github|ziy|1.3.1

(4 theme extensions excluded)

</details>

**Comment by forivall** [comment]
@vscodebot it's not. thanks though, ya little robot!

**Comment by egamma** [comment]
Fixed as part of #54085
